spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -289,6 +292,19 @@ public final void initialize(String name, CaseInsensitiveStringMap options) {
     this.createOrcAsIceberg = options.getBoolean("orc-enabled", createOrcAsIceberg);
   }
 
+  private void checkHiveSupport() {
+    RuntimeConfig runtimeConfig = SparkSession.active().conf();
+    String sparkCatalogImpl = runtimeConfig.get(StaticSQLConf.CATALOG_IMPLEMENTATION().key());
+    if (sparkCatalogImpl != null) {
+      Preconditions.checkArgument(
+          sparkCatalogImpl.equals("hive"),
+          "Please enable hive support for Spark. "
```

Comment:
should the error msg mention how to enable it? `Please enable hive support for Spark by setting 'spark.sql.catalogImplementation=hive'` or something along those lines

Reply:
> should the error msg mention how to enable it? `Please enable hive support for Spark by setting 'spark.sql.catalogImplementation=hive'` or something along those lines

How about `Please enable hive support for Spark by calling enableHiveSupport() or setting 'spark.sql.catalogImplementation=hive'`

Reply:
sgtm